daemon/openrazer_daemon/dbus_services/dbus_methods/mamba.py

```diff
@@ -312,8 +312,17 @@ def set_poll_rate(self, rate):
     """
     self.logger.debug("DBus call set_poll_rate")
 
+    # Modified to allow devices with a difference in allowed polling rates
+    # between wired/wireless modes to not crash daemon on mode change
+
+    # Should probably make system to store previous rates by Product ID
     if rate not in self.POLL_RATES:
-        raise RuntimeError("Poll rate " + str(rate) + " is not allowed. Allowed values: " + str(self.POLL_RATES))
+        if rate > self.POLL_RATES[-1]:
+            self.logger.warning("Poll rate " + str(rate) + " is above allowed values: " + str(self.POLL_RATES) + ". Setting to max safe value")
+            rate = self.POLL_RATES[-1]
+        else:
+            self.logger.warning("Poll rate " + str(rate) + " is outside of Allowed values: " + str(self.POLL_RATES) + ". Setting to lowest value")
+            rate = self.POLL_RATES[1]
```

Comment:
I don't quite get what should be fixed with this? If this method is being called from an external application with wrong parameters it's correct to throw an exception.

If it's something else, please post the stack trace or a way to reproduce.

Reply:
I can definitely provide reproduction steps. My environment is an Ubuntu 22.04 VM with just the openrazer driver and polychromatic installed. 

1. Connect mouse in wireless mode
2. Set polling rate above the max in wired mode (Say 8000Hz).
3. Unplug wireless dongle and connect via USB cable.
4. Observe daemon throw exception and become unresponsive.

After doing some testing, this function seems to be called on initialization of the mouse. It seems to happen regardless of if the front end is running or not. Furthermore, it will keep trying to set an incorrect polling rate on each subsequent daemon restart, including after a reboot. The only way to correct this from a users perspective is to re-connect the mouse wirelessly, restart the daemon, and set the polling rate <= the max wired polling rate.

This may be some issue specific to this mouse, or maybe something I've inadvertently created. I'm not sure.

If you'd like me to revert this back to the previous configuration. Please let me know.

<details>
<summary>Daemon debug output, including python traceback.</summary>

~~~
Starting daemon.
2023-08-19 07:24:44 | razer                          | INFO     | Initialising Daemon (v3.6.1). Pid: 6976
2023-08-19 07:24:44 | razer.screensaver              | INFO     | Initialising DBus Screensaver Monitor
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAbyssus ---------------------------- (1532:0042)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAbyssus1800 ------------------------ (1532:0020)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAbyssus2000 ------------------------ (1532:005E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAbyssusEliteDVaEdition ------------- (1532:006A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAbyssusEssential ------------------- (1532:006B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAbyssusV2 -------------------------- (1532:005B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAnansi ----------------------------- (1532:010F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerAtherisReceiver -------------------- (1532:0062)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBaseStationChroma ------------------ (1532:0F08)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBaseStationV2Chroma ---------------- (1532:0F20)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasilisk --------------------------- (1532:0064)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskEssential ------------------ (1532:0065)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskUltimateReceiver ----------- (1532:0088)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskUltimateWired -------------- (1532:0086)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskV2 ------------------------- (1532:0085)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskV3 ------------------------- (1532:0099)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskV3ProWired ----------------- (1532:00AA)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskV3ProWireless -------------- (1532:00AB)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBasiliskXHyperSpeed ---------------- (1532:0083)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidow2019 --------------------- (1532:0241)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowChroma ------------------- (1532:0203)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowChromaOverwatch ---------- (1532:0211)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowChromaTournamentEdition -- (1532:0209)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowChromaV2 ----------------- (1532:0221)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowElite -------------------- (1532:0228)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowEssential ---------------- (1532:0237)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowLite --------------------- (1532:0235)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowStealth ------------------ (1532:011B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowStealthEdition ----------- (1532:010E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowTournamentEdition2014 ---- (1532:011C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowUltimate2012 ------------- (1532:010D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowUltimate2013 ------------- (1532:011A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowUltimate2016 ------------- (1532:0214)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowV3 ----------------------- (1532:024E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowV3MiniHyperspeed --------- (1532:0258)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowV3MiniHyperspeedWireless - (1532:0271)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowV3ProWired --------------- (1532:025A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowV3TK --------------------- (1532:0A24)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowXChroma ------------------ (1532:0216)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowXTournamentEditionChroma - (1532:021A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlackWidowXUltimate ---------------- (1532:0217)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade142021 ------------------------ (1532:0270)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade142022 ------------------------ (1532:028C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade15Advanced2020 ---------------- (1532:0253)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade15Advanced2021 ---------------- (1532:0276)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade15AdvancedEarly2021 ----------- (1532:026D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade15AdvancedEarly2022 ----------- (1532:028A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade172022 ------------------------ (1532:028B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade17Pro2021 --------------------- (1532:0279)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade17ProEarly2021 ---------------- (1532:026E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade2018 -------------------------- (1532:0233)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade2018Base ---------------------- (1532:023B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade2018Mercury ------------------- (1532:0240)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade2019Adv ----------------------- (1532:023A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade2019Base ---------------------- (1532:0246)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBlade2019StudioEdition ------------- (1532:024D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeEarly2020Base ----------------- (1532:0255)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeEarly2021Base ----------------- (1532:026F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeEarly2022Base ----------------- (1532:027A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeLate2016 ---------------------- (1532:0224)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeMid2019Mercury ---------------- (1532:0245)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladePro2017 ----------------------- (1532:0225)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladePro2017FullHD ----------------- (1532:022F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladePro2019 ----------------------- (1532:0234)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeProEarly2020 ------------------ (1532:0256)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeProLate2016 ------------------- (1532:0210)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeProLate2019 ------------------- (1532:024C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeQHD --------------------------- (1532:020F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealth ----------------------- (1532:0205)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealth2019 ------------------- (1532:0239)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealthEarly2020 -------------- (1532:0252)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealthLate2016 --------------- (1532:0220)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealthLate2017 --------------- (1532:0232)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealthLate2019 --------------- (1532:024A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealthLate2020 --------------- (1532:0259)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBladeStealthMid2017 ---------------- (1532:022D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerBook2020 --------------------------- (1532:026A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerChargingPadChroma ------------------ (1532:0F26)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerChromaARGB ------------------------- (1532:0F1F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerChromaHDK -------------------------- (1532:0F09)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerChromaMugHolder -------------------- (1532:0F07)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerCore ------------------------------- (1532:0215)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerCoreXChroma ------------------------ (1532:0F1A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerCynosaChroma ----------------------- (1532:022A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerCynosaChromaPro -------------------- (1532:022C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerCynosaLite ------------------------- (1532:023F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerCynosaV2 --------------------------- (1532:025E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdder1800 --------------------- (1532:0038)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdder2000 --------------------- (1532:004F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdder2013 --------------------- (1532:0037)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdder3500 --------------------- (1532:0054)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdder3_5G --------------------- (1532:0016)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdder3_5GBlack ---------------- (1532:0029)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderChroma ------------------- (1532:0043)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderElite -------------------- (1532:005C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderEssential ---------------- (1532:006E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderEssential2021 ------------ (1532:0098)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderEssentialWhiteEdition ---- (1532:0071)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV2 ----------------------- (1532:0084)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV2Lite ------------------- (1532:00A1)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV2Mini ------------------- (1532:008C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV2ProWired --------------- (1532:007C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV2ProWireless ------------ (1532:007D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV2XHyperSpeed ------------ (1532:009C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV3ProWired --------------- (1532:00B6)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathAdderV3ProWireless ------------ (1532:00B7)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathStalkerChroma ----------------- (1532:0204)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathStalkerExpert ----------------- (1532:0202)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathStalkerV2 --------------------- (1532:0295)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathStalkerV2ProTKLWired ---------- (1532:0298)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathStalkerV2ProTKLWireless ------- (1532:0296)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathStalkerV2ProWired ------------- (1532:0292)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDeathStalkerV2ProWireless ---------- (1532:0290)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerDiamondbackChroma ------------------ (1532:004C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerFirefly ---------------------------- (1532:0C00)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerFireflyHyperflux ------------------- (1532:0068)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerFireflyV2 -------------------------- (1532:0C04)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerGoliathus -------------------------- (1532:0C01)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerGoliathusExtended ------------------ (1532:0C02)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsman --------------------------- (1532:0227)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanElite ---------------------- (1532:0226)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanMini ----------------------- (1532:0257)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanMiniAnalog ----------------- (1532:0282)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanMiniJP --------------------- (1532:0269)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanTournamentEdition ---------- (1532:0243)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanV2 ------------------------- (1532:026C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanV2Analog ------------------- (1532:0266)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHuntsmanV2Tenkeyless --------------- (1532:026B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerHyperPollingWirelessDongle --------- (1532:00B3)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerImperator -------------------------- (1532:002F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerKraken71 --------------------------- (1532:0501)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerKraken71Alternate ------------------ (1532:0506)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerKraken71Chroma --------------------- (1532:0504)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerKraken71V2 ------------------------- (1532:0510)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerKrakenKittyEdition ----------------- (1532:0F19)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerKrakenUltimate --------------------- (1532:0527)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerLanceheadTE ------------------------ (1532:0060)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerLanceheadWired --------------------- (1532:0059)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerLanceheadWireless ------------------ (1532:005A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerLanceheadWirelessReceiver ---------- (1532:006F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerLanceheadWirelessWired ------------- (1532:0070)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerLaptopStandChroma ------------------ (1532:0F0D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerLaptopStandChromaV2 ---------------- (1532:0F2B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMamba2012Wired --------------------- (1532:0024)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMamba2012Wireless ------------------ (1532:0025)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMambaChromaWired ------------------- (1532:0044)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMambaChromaWireless ---------------- (1532:0045)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMambaElite ------------------------- (1532:006C)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMambaTE ---------------------------- (1532:0046)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMambaWirelessReceiver -------------- (1532:0072)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMambaWirelessWired ----------------- (1532:0073)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMouseBungeeV3Chroma ---------------- (1532:0F1D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerMouseDock -------------------------- (1532:007E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNaga2012 --------------------------- (1532:002E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNaga2014 --------------------------- (1532:0040)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaChroma ------------------------- (1532:0053)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaEpicChromaWired ---------------- (1532:003E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaEpicChromaWireless ------------- (1532:003F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaHex ---------------------------- (1532:0041)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaHexRed ------------------------- (1532:0036)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaHexV2 -------------------------- (1532:0050)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaLeftHanded2020 ----------------- (1532:008D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaProWired ----------------------- (1532:008F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaProWireless -------------------- (1532:0090)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaTrinity ------------------------ (1532:0067)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNagaX ------------------------------ (1532:0096)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNommoChroma ------------------------ (1532:0517)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNommoPro --------------------------- (1532:0518)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerNostromo --------------------------- (1532:0111)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrbweaver -------------------------- (1532:0113)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrbweaverChroma -------------------- (1532:0207)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrnata ----------------------------- (1532:021F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrnataChroma ----------------------- (1532:021E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrnataV2 --------------------------- (1532:025D)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrnataV3X -------------------------- (1532:0294)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrochi2011 ------------------------- (1532:0013)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrochi2013 ------------------------- (1532:0039)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrochiV2Bluetooth ------------------ (1532:0095)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrochiV2Receiver ------------------- (1532:0094)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOrochiWired ------------------------ (1532:0048)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerOuroboros -------------------------- (1532:0032)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerProClickMiniReceiver --------------- (1532:009A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerProClickReceiver ------------------- (1532:0077)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerProClickWired ---------------------- (1532:0080)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerRaptor27 --------------------------- (1532:0F12)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerTaipan ----------------------------- (1532:0034)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerTartarus --------------------------- (1532:0201)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerTartarusChroma --------------------- (1532:0208)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerTartarusV2 ------------------------- (1532:022B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerThunderbolt4DockChroma ------------- (1532:0F21)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViper ------------------------------ (1532:0078)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViper8KHz -------------------------- (1532:0091)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViperMini -------------------------- (1532:008A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViperMiniSEWired ------------------- (1532:009E)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViperMiniSEWireless ---------------- (1532:009F)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViperUltimateWired ----------------- (1532:007A)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViperUltimateWireless -------------- (1532:007B)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViperV2ProWired -------------------- (1532:00A5)
2023-08-19 07:24:44 | razer                          | DEBUG    | Loaded device specification: RazerViperV2ProWireless ----------------- (1532:00A6)
2023-08-19 07:24:44 | razer                          | INFO     | Found device.0: 0003:1532:009F.0008
2023-08-19 07:24:44 | razer.device0                  | INFO     | Initialising device.0 RazerViperMiniSEWireless
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getRazerUrls method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.lighting.chroma.restoreLastEffect method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceImage method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.suspendDevice method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceMode method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getSerial method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.setDeviceMode method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getVidPid method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDriverVersion method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.hasDedicatedMacroKeys method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.resumeDevice method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getFirmware method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getMatrixDimensions method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.hasMatrix method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceName method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceType method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.dpi.maxDPI method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.dpi.getDPI method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.dpi.setDPI method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.dpi.getDPIStages method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.dpi.setDPIStages method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getPollRate method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.setPollRate method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.getSupportedPollRates method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.power.getBattery method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.power.isCharging method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.power.getIdleTime method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.power.setIdleTime method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.power.setLowBatteryThreshold method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Adding razer.device.misc.setHyperPollingLED method to DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | DBus call set_dpi_xy
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Set persistence (None, dpi_x, 1800)
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | Set persistence (None, dpi_y, 1800)
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | DBus call set_poll_rate
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.devices.getDevices method to DBus
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.devices.enableTurnOffOnScreensaver method to DBus
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.devices.getSyncEffects method to DBus
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.daemon.stop method to DBus
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.devices.getOffOnScreensaver method to DBus
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.devices.syncEffects method to DBus
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.daemon.version method to DBus
2023-08-19 07:24:44 | razer                          | DEBUG    | Adding razer.devices.supportedDevices method to DBus
2023-08-19 07:24:44 | razer                          | INFO     | Serving DBus
2023-08-19 07:24:44 | razer.device0                  | DEBUG    | DBus call get_battery
2023-08-19 07:24:44 | razer.device0.batterynotifier  | DEBUG    | Razer Viper Mini Signature Edition Battery at 81%
2023-08-19 07:24:54 | razer                          | DEBUG    | State recently changed, writing to disk
2023-08-19 07:24:54 | razer                          | DEBUG    | Writing persistence config
2023-08-19 07:24:56 | razer                          | DEBUG    | DBus called get_serial_list
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_device_type
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_device_name
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_device_type
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_firmware
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call version
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call has_matrix
2023-08-19 07:24:56 | razer                          | DEBUG    | DBus called get_serial_list
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_device_type
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_device_name
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_device_type
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_firmware
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call version
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call has_matrix
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call max_dpi
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call is_charging
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_battery
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_poll_rate
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_supported_poll_rates
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_poll_rate
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_idle_time
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_dpi_xy
2023-08-19 07:24:56 | razer.device0                  | DEBUG    | DBus call get_dpi_xy
2023-08-19 07:25:02 | razer.device0                  | DEBUG    | DBus call set_poll_rate
2023-08-19 07:25:28 | razer                          | DEBUG    | Device event [unbind]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-4/1-4:1.0/0003:1532:009F.0008
2023-08-19 07:25:28 | razer                          | DEBUG    | Device event [remove]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-4/1-4:1.0/0003:1532:009F.0008
2023-08-19 07:25:28 | razer.device0                  | DEBUG    | DBus call get_dpi_xy
2023-08-19 07:25:28 | razer.device0.batterymanager   | DEBUG    | Closing Battery Manager
2023-08-19 07:25:29 | razer.device0.batterynotifier  | DEBUG    | Shutting down battery notifier
2023-08-19 07:25:29 | razer                          | DEBUG    | Writing persistence config
2023-08-19 07:25:29 | razer                          | WARNING  | Removing 0003:1532:009F.0008
2023-08-19 07:25:29 | razer                          | DEBUG    | Emitted Device Remove Signal
2023-08-19 07:25:29 | razer                          | DEBUG    | Device event [unbind]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-4/1-4:1.1/0003:1532:009F.0009
2023-08-19 07:25:29 | razer                          | DEBUG    | Device event [remove]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-4/1-4:1.1/0003:1532:009F.0009
2023-08-19 07:25:29 | razer                          | DEBUG    | Device event [unbind]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-4/1-4:1.2/0003:1532:009F.000A
2023-08-19 07:25:29 | razer                          | DEBUG    | Device event [remove]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-4/1-4:1.2/0003:1532:009F.000A
2023-08-19 07:25:43 | razer                          | DEBUG    | Device event [add]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-5/1-5:1.2/0003:1532:009E.000D
2023-08-19 07:25:43 | razer                          | DEBUG    | Device event [add]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-5/1-5:1.0/0003:1532:009E.000B
2023-08-19 07:25:43 | razer                          | DEBUG    | Device event [add]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-5/1-5:1.1/0003:1532:009E.000C
2023-08-19 07:25:43 | razer                          | DEBUG    | Device event [bind]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-5/1-5:1.2/0003:1532:009E.000D
2023-08-19 07:25:43 | razer                          | DEBUG    | Device event [bind]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-5/1-5:1.1/0003:1532:009E.000C
2023-08-19 07:25:43 | razer                          | DEBUG    | Device event [bind]: /devices/pci0000:00/0000:00:02.1/0000:02:00.0/usb1/1-5/1-5:1.0/0003:1532:009E.000B
2023-08-19 07:25:45 | razer                          | INFO     | Found valid device.0: 0003:1532:009E.000B
2023-08-19 07:25:45 | razer.device0                  | INFO     | Initialising device.0 RazerViperMiniSEWired
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.lighting.chroma.restoreLastEffect method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getRazerUrls method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceMode method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.setDeviceMode method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.suspendDevice method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.resumeDevice method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceImage method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getVidPid method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.hasDedicatedMacroKeys method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getSerial method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDriverVersion method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getFirmware method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getMatrixDimensions method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.hasMatrix method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceName method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getDeviceType method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.dpi.maxDPI method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.dpi.getDPI method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.dpi.setDPI method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.dpi.getDPIStages method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.dpi.setDPIStages method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getPollRate method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.setPollRate method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.misc.getSupportedPollRates method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.power.getBattery method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.power.isCharging method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.power.getIdleTime method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.power.setIdleTime method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Adding razer.device.power.setLowBatteryThreshold method to DBus
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | DBus call set_dpi_xy
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Set persistence (None, dpi_x, 1800)
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | Set persistence (None, dpi_y, 1800)
2023-08-19 07:25:45 | razer.device0                  | DEBUG    | DBus call set_poll_rate
Exception in thread Thread-3 (_collecting_udev_method):
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3/dist-packages/openrazer_daemon/daemon.py", line 605, in _collecting_udev_method
    self._add_device(d)
  File "/usr/lib/python3/dist-packages/openrazer_daemon/daemon.py", line 530, in _add_device
    razer_device = device_class(device_path=sys_path, device_number=device_number, config=self._config,
  File "/usr/lib/python3/dist-packages/openrazer_daemon/hardware/mouse.py", line 1574, in __init__
    super().__init__(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/openrazer_daemon/hardware/device_base.py", line 316, in __init__
    self.restore_dpi_poll_rate()
  File "/usr/lib/python3/dist-packages/openrazer_daemon/hardware/device_base.py", line 356, in restore_dpi_poll_rate
    poll_rate_func(self.poll_rate)
  File "/usr/lib/python3/dist-packages/openrazer_daemon/dbus_services/dbus_methods/mamba.py", line 324, in set_poll_rate
    raise RuntimeError("Poll rate " + str(rate) + " is not allowed. Allowed values: " + str(self.POLL_RATES))
RuntimeError: Poll rate 8000 is not allowed. Allowed values: [125, 500, 1000]
2023-08-19 07:25:54 | razer                          | DEBUG    | State recently changed, writing to disk
2023-08-19 07:25:54 | razer                          | DEBUG    | Writing persistence config
2023-08-19 07:26:49 | razer                          | DEBUG    | DBus called get_serial_list
2023-08-19 07:26:49 | razer                          | DEBUG    | DBus called get_serial_list
~~~

</details>

Reply:
Ah I think I get it.

Wired & wireless version have the same serial number (I imagine). The daemon stores in the persistence config what polling rate the device with serial X is set to. When you reconnect the device but daemon thinks it cannot to the 8kHz, then it'll fail.

But I think then instead of clamping the values here, this should be done in the persistence restore code, to check if it can still do X Hz or if it has to lower (or increase in some other weird device) the polling rate to the max supported one. I can also look into this if you don't find where to do this or a good way to do this.

Reply:
Gotcha, I think you've nailed the issue. I'll see what I can do about fixing it. In the meantime I've reverted set_poll_rate back to normal.

You're absolutely right on the serial numbers by the way. Reading /sys/bus/hid/drivers/razermouse/{IDENTIFIER}/device_serial in both wired/wireless modes gives the same value.

Since this is most likely a different problem, would you like me to open a different issue?

Reply:
Should be resolved with https://github.com/openrazer/openrazer/commit/1cad208ff3df2349fdbbf09f16bcd0df7808dcb1 , please try :)

Reply:
It works perfect! Thanks for implementing that so quickly.

Reply:
Well, thanks for reporting it in a clear way!